services/gateway/gateway-service.mjs

```diff
@@ -339,11 +373,54 @@ class GatewayService {
       console.log('player has no last time record')
       return false;
     }
-    if( Date.now() - playerData.lastMovingTime < moveingRequestThreshold){
+    if( Date.now() - playerData.lastMovingTime < movingRequestThreshold){
       return false;
     }
     return true;
   }
+
```

Comment:
Where is the obstacle check? You should block players if they try to walk across a wall.

Reply:
I'll leave this to whyang.